### DISCUSSION TITLE
Running YouTubeSpammerPurge.py on a Mac

### DISCUSSION THREAD

**Discussion by AudioKev** [ThioJoe/YT-Spammer-Purge#281] [Q&A]
Greetings,

While following the instructions specific to Mac OS, the following were encountered:

DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621

The previous "note" occurred 26 times while "pip3 install -r requirements.txt" executed. Does it mean that something did not install or install correctly? Might this be why the program failed, as shown below. (I used Homebrew for a previous project.)

When attempting to run the Purge program, the following was the result:

AudioKev@Mac-mini YT-Spammer-Purge-2.4.2 % python3 YouTubeSpammerPurge.py  
Traceback (most recent call last):
  File "/.../Downloads/YT-Spammer-Purge-2.4.2/YouTubeSpammerPurge.py", line 43, in <module>
    from gui import *
  File "/.../Downloads/YT-Spammer-Purge-2.4.2/gui.py", line 5, in <module>
    import YouTubeSpammerPurge as myFunctions
  File "/.../Downloads/YT-Spammer-Purge-2.4.2/YouTubeSpammerPurge.py", line 54, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'

Any help is greatly appreciated. Thanks.

**Comment by rbeer** [comment]
`pip3 install requests`

**Reply by AudioKev** [reply]
Thank you for responding so quickly.

Here is what happened when issuing the pip3 command as suggested:

AudioKev@Mac-mini YT-Spammer-Purge-2.4.2 % pip3 install requests         
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Requirement already satisfied: requests in /usr/local/lib/python3.9/site-packages (2.26.0)
Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.9/site-packages (from requests) (2.0.9)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/site-packages (from requests) (1.26.7)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.9/site-packages (from requests) (2021.10.8)
Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.9/site-packages (from requests) (3.3)

The results of running the python3 YouTubeSpammerPurge.py command is as before, ending is with "ModuleNotFoundError: No module named 'requests'".

**Reply by Kurdakov** [reply]
It's probably due to you have two python3 installed (one new and another from brew).
try steps https://programmerah.com/solved-modulenotfounderror-no-module-named-requests-30575/ to fix the problem

**Reply by TechStudent10** [reply]
@AudioKev Run each of these commands then `python3 YouTubeSpammerPurge.py` or `python YouTubeSpammerPurge.py` and see if any of them works:

`pip install -U requests`
`pip3 install -U requests`
`pip install requests`
`pip3 install requests`
`python -m pip install requests`
`python3 -m pip install requests`
`python -m pip3 install requests`
`python3 -m pip3 install requests`
`python -m pip install -U requests`
`python3 -m pip install -U requests`
`python -m pip3 install -U requests`
`python3 -m pip3 install -U requests`
`python -m pip install requests`
`python3 -m pip install requests`
`python -m pip3 install requests`
`python3 -m pip3 install requests`
`python -m pip install -U requests`
`python3 -m pip install -U requests`
`python -m pip3 install -U requests`
`python3 -m pip3 install -U requests`
`py -m pip install requests`
`py -m pip3 install requests`
`py -m pip install -U requests`
`py -m pip3 install -U requests`

_I just listed **EVERY POSSIBLE WAY** (that I know of) to install a Python module_

And as for the depreciation notice, don't worry about that. That's just telling you that a module is, well, deprecated, and will be no longer work **in the near future**.

**Reply by Kurdakov** [reply]
> `python YouTubeSpammerPurge.py` and see if any of them works:

OSX has python 2.7 installed by default so python YouTubeSpammerPurge.py likely will call pyton2.7 and will crash somewhere

**Reply by TechStudent10** [reply]
@Kurdakov that's true. @AudioKev install Python 3 and see if it works.

**Reply by rbeer** [reply]
> It's probably due to you have two python3 installed (one new and another from brew).
Would be my next guess, at this point.

You can do `python3 -V` and `pip3 -V` (capital V) to take a quick look at the versions. The `pip3 -V` tells you what python version it uses.

`Requirement already satisfied: requests in /usr/local/lib/python3.9/site-packages (2.26.0)` tells you where and for what version `pip3` has installed modules; `/usr/local/lib/python3.9`, in this case. You might have another `pythonX.Y` folder in `/usr/local/lib`, just take a look with `ls`. Beyond that, the link @Kurdakov posted looks good.

**Reply by KendallDoesCoding** [reply]
Hope your issue is fixed @AudioKev

**Reply by AudioKev** [reply]
Back from other projects.

I have yet to correct the issue; SpammerPurge remains inoperative on my Mac.

The latest problem:

AudioKev@Mac-mini YT-Spammer-Purge-2.4.2 % python3 YouTubeSpammerPurge.py
Traceback (most recent call last):
  File "/Users/AudioKev/Downloads/YT-Spammer-Purge-2.4.2/YouTubeSpammerPurge.py", line 43, in <module>
    from gui import *
  File "/Users/AudioKev/Downloads/YT-Spammer-Purge-2.4.2/gui.py", line 5, in <module>
    import YouTubeSpammerPurge as myFunctions
  File "/Users/AudioKev/Downloads/YT-Spammer-Purge-2.4.2/YouTubeSpammerPurge.py", line 69, in <module>
    from googleapiclient.errors import HttpError
ModuleNotFoundError: No module named '**googleapiclient**'

AudioKev@Mac-mini YT-Spammer-Purge-2.4.2 % sudo pip3 install **googleapiclient** 
WARNING: The directory '/Users/AudioKev/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
ERROR: Could not find a version that satisfies the requirement **googleapiclient** (from versions: none)
ERROR: No matching distribution found for **googleapiclient**

As an alternative, I successfully ran SpammerPurge on a notebook running Windows 10. Though this worked, I would still like to have it operational on my Mac.

**Reply by TechStudent10** [reply]
First, don't run with `sudo` if you don't have to. Second, try installing the requirements again.

**Reply by AudioKev** [reply]
I appreciate your help!

Neither of the suggestions helped; pip3 install googleapiclient (without sudo) or an attempt to reinstall the files listed in the requirements.txt file (everything was "already satisfied."

Mystery ....

**Reply by rbeer** [reply]
> ERROR: No matching distribution found for **googleapiclient**

This is not what the module is called. You want `google_api_python_client`. Look into requirements.txt or use `pip3 -r requirements` to install the dependencies.

**Reply by Kurdakov** [reply]
I think the option is to completely remove  previous homebrew python  (if you don't need it)
brew uninstall python3
maybe even uninstall all python3 installations and check python3 in command line doesn't work
